draft-ietf-jsonpath-base.md

```diff
@@ -902,11 +965,173 @@ size doesn't
 fit in the implementation's representation of an integer, the implementation
 MUST raise an error.
 
+### Descendant Selector
+
+#### Syntax
+{: unnumbered}
+
+The descendant selector starts with a double dot `..` and can be
+followed by an object member name (similar to the `dot-selector`),
+by an `index-selector` acting on objects or arrays, or by a wild card.
+
+~~~~ abnf
+descendant-selector = ".." ( dot-member-name      /  ; ..<name>
+                             index-selector       /  ; ..[<index>]
+                             index-wild-selector  /  ; ..[*]
+                             "*"                     ; ..*
+                           )
+~~~~
+
+#### Semantics
+{: unnumbered}
+
+The `descendant-selector` is inspired by ECMAScript for XML (E4X). It
+selects the nodes of object member values and array elements in a recursive descent process.
```

Comment:
The descendant selector should select all primitive values which are part of the subtree of the input node. For example, `..*` applied to the argument `true` should produce `[true]` even though there are no object members or array elements in the argument.

The descendant selector should also select _all_ object members and _all_ array elements (the current wording could permit some to be omitted).
```suggestion
It selects the node and all its descendants..
```

Reply:
To do: Define "descendants" (making sure that member values are, but member names aren't).